compiler/src/iree/compiler/PluginAPI/PluginManager.h

```diff
@@ -66,29 +69,44 @@ class PluginManager : public PluginRegistrar {
 
   // Calls through to AbstractPluginRegistration::registerDialects for all
   // available plugins.
-  void registerDialects(DialectRegistry &registry);
+  void registerGlobalDialects(DialectRegistry &registry);
 
  private:
   friend class PluginManagerSession;
 };
 
 // Holds activated plugins for an |iree_compiler_session_t|.
-class PluginManagerSession {
+class PluginManagerSession : public PipelineExtensions {
  public:
   PluginManagerSession(PluginManager &pluginManager, OptionsBinder &binder,
                        PluginManagerOptions &options);
 
+  // Initializes all plugins that should be activated by default.
+  LogicalResult initializePlugins();
+
+  // Invokes registerDialects() on all initialized plugins.
```

Comment:
Function name gave me pause ... But don't have better suggestions and it would be clear if used as member function.